### PR TITLE
feat: Add NPM Action

### PR DESCRIPTION
**Pull request by kylemellander**
The yarn action allows for setting up node_modules with caching while using npm and package-lock.json.

### CODE REVIEW

File: npm/action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-node@v2
+      with:
+        node-version: ${{ inputs.nodeVersion }}
+        cache: 'npm'
```
Comment (sheck) [inline npm/action.yml:12]: Do we still want the additional caching here, or should it be removed like we did with the yarn action?

**Review by sheck — APPROVED** [review]

File: npm/action.yml (hunk 1 above)
Comment (kylemellander) [reply npm/action.yml:12]: I'll run a test to see how it impacts

Comment (kylemellander) [reply npm/action.yml:12]: I tested it out and it was actually slightly faster leaving the caching in.  I'm ok with removing it for the sake of consistency, but it does not add the slowdown like the yarn cache did.

Comment (sheck) [reply npm/action.yml:12]: Sounds good, feel free to merge it when you're ready

**Pull request merged**